pwiz_tools/BiblioSpec/src/DiaNNSpecLibReader.cpp

```diff
@@ -669,11 +675,16 @@ bool DiaNNSpecLibReader::parseFile()
 
             auto findItr = speclib.entryByModPeptideAndCharge.find(precursorId);
             if (findItr == speclib.entryByModPeptideAndCharge.end())
+            {
+                // skip contaminant proteins if they are not included in the speclib file
+                if (bal::starts_with(proteinGrp, "contaminant_"))
```

Comment:
Seems pretty arbitrary. This is totally dependent on what's in the FASTA file, right?

Reply:
I added it to be able to process the dataset from the Van Eyk lab. Otherwise, it stops with an error. I suppose another option would be to just skip anything that is missing. This is at least limited to a known case for now. Can it hurt?

Reply:
I can't find where in DIA-NN it has any special handling for contaminant proteins. Maybe it's hard-coded (but I checked the last open source version of diann.cpp and it's not there). @vdemichev could you please comment on how precursorIds could be in the tsv but not the speclib?